src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PeriodicBatchingSink.cs

```diff
@@ -56,305 +58,209 @@ public class PeriodicBatchingSink : ILogEventSink, IDisposable, IBatchedLogEvent
-    }
-
-    PeriodicBatchingSink(PeriodicBatchingSinkOptions options)
     {
         if (options == null) throw new ArgumentNullException(nameof(options));
-
         if (options.BatchSizeLimit <= 0)
             throw new ArgumentOutOfRangeException(nameof(options), "The batch size limit must be greater than zero.");
         if (options.Period <= TimeSpan.Zero)
             throw new ArgumentOutOfRangeException(nameof(options), "The period must be greater than zero.");
 
+        _targetSink = batchedSink ?? throw new ArgumentNullException(nameof(batchedSink));
         _batchSizeLimit = options.BatchSizeLimit;
-        _queue = new(options.QueueLimit);
-        _status = new(options.Period);
+        _queue = options.QueueLimit is { } limit
+            ? Channel.CreateBounded<LogEvent>(new BoundedChannelOptions(limit) { SingleReader = true })
+            : Channel.CreateUnbounded<LogEvent>(new UnboundedChannelOptions { SingleReader = true });
+        _batchScheduler = new FailureAwareBatchScheduler(options.Period);
         _eagerlyEmitFirstEvent = options.EagerlyEmitFirstEvent;
-        _timer = new(_ => OnTick());
+        _waitForShutdownSignal = Task.Delay(Timeout.InfiniteTimeSpan, _shutdownSignal.Token)
+            .ContinueWith(e => e.Exception, TaskContinuationOptions.OnlyOnFaulted);
 
-        // Initialized by externally-callable constructors.
-        _batchedLogEventSink = null!;
+        _runLoop = Task.Run(LoopAsync);
     }
 
     /// <summary>
-    /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+    /// Emit the provided log event to the sink. If the sink is being disposed or
+    /// the app domain unloaded, then the event is ignored.
     /// </summary>
-    /// <filterpriority>2</filterpriority>
-    public void Dispose()
+    /// <param name="logEvent">Log event to emit.</param>
+    /// <exception cref="ArgumentNullException">The event is null.</exception>
+    /// <remarks>
+    /// The sink implements the contract that any events whose Emit() method has
+    /// completed at the time of sink disposal will be flushed (or attempted to,
+    /// depending on app domain state).
+    /// </remarks>
+    public void Emit(LogEvent logEvent)
     {
-        Dispose(true);
-        GC.SuppressFinalize(this);
-    }
+        if (logEvent == null) throw new ArgumentNullException(nameof(logEvent));
 
-    /// <summary>
-    /// Free resources held by the sink.
-    /// </summary>
-    /// <param name="disposing">If true, called because the object is being disposed; if false,
-    /// the object is being disposed from the finalizer.</param>
-    protected virtual void Dispose(bool disposing)
-    {
-        if (!disposing) return;
+        if (_shutdownSignal.IsCancellationRequested)
+            return;
 
-        lock (_stateLock)
+        _queue.Writer.TryWrite(logEvent);
+    }
+    
+    /// <inheritdoc/>
+    public void Dispose()
```

Comment:
I know the diffs will look bad, but the Dispose stuff needs to move out of here to the end!